sdk/python/kfp/_client.py

```diff
@@ -307,12 +307,12 @@ def get_kfp_healthz(self):
     while not response:
       count += 1
       if count > max_attempts:
-        raise TimeoutError('API call timeout')
+        raise TimeoutError('Failed getting healthz endpoint after {} attempts.'.format(max_attempts))
       try:
         response = self._healthz_api.get_healthz()
         return response
-      except:
-        logging.info('Failed to get healthz info attempt {} of 5.'.format(count))
+      except Exception:
```

Comment:
it seems catching `Exception` instead of all is much better: https://stackoverflow.com/a/4992124/8745218

e.g. it won't catch system exit signal nor keyboard interrupt exception

Reply:
Yes, it's always preferable to catch the most specific exception first. I took a quick check, and if I'm not mistaken the more specific error is likely `ApiException`:

https://github.com/kubeflow/pipelines/blob/1449cfe0a5efeaeba9f5b6d26ce1d58024657814/backend/api/python_http_client/kfp_server_api/api_client.py#L193-L202

It may also throw `ApiValueError`, but that's not something we want to sleep and retry here.

Reply:
Agree that we should not retry with ApiException.

Reply:
Thanks, will update